Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/CorelliPowderCalibrationCreate.py

```diff
@@ -0,0 +1,576 @@
+        return issues
+
+    def PyInit(self):
+        self.declareProperty(
+            IEventWorkspaceProperty('InputWorkspace', '',
+                                    direction=Direction.Input,
+                                    validator=WorkspaceUnitValidator('TOF')),
+            doc='Powder event data, ideally from a highly symmetric space group',
+        )
+        self.declareProperty(name='OutputWorkspacesPrefix', defaultValue='pdcal_', direction=Direction.Input,
+                             doc="Prefix to be added to output workspaces")
+        # Tube Calibration properties
+        self.declareProperty(name='TubeDatabaseDir', defaultValue='', direction=Direction.Input,
+                             doc='path to database containing detector heights')
+        # PDCalibration properties exposed, grouped
+        property_names = ['TofBinning', 'PeakFunction', 'PeakPositions']
+        self.copyProperties('PDCalibration', property_names)
+        [self.setPropertyGroup(name, 'PDCalibration') for name in property_names]
+        # AlignComponents properties exposed, grouped
+        property_names = ['AdjustSource', 'SourceMaxTranslation', 'ComponentList', 'ComponentMaxTranslation',
+                          'ComponentMaxRotation']
+        self.declareProperty(name='AdjustSource', defaultValue=True,
+                             doc='Adjust Z-coordinate of the source')
+        self.declareProperty(name='SourceMaxTranslation', defaultValue=0.1,
+                             doc='Maximum adjustment of source position along the beam (Z) axis (m)')
+        self.declareProperty(StringArrayProperty('ComponentList', values=self._banks, direction=Direction.Input),
+                             doc='Comma separated list on banks to refine')
+        self.declareProperty(name='ComponentMaxTranslation', defaultValue=0.02,
+                             doc='Maximum translation of each component along either of the X, Y, Z axes (m)')
+        self.declareProperty(name='ComponentMaxRotation', defaultValue=3.0,
+                             doc='Maximum rotation of each component along either of the X, Y, Z axes (deg)')
+        [self.setPropertyGroup(name, 'AlignComponents') for name in property_names]
+
+    def PyExec(self):
+        temporary_workspaces = []
+        self.temp_ws = temp_workspace_generator(temporary_workspaces)  # name generator for temporary workpaces
+
+        prefix_output = self.getProperty('OutputWorkspacesPrefix').value
+        progress_percent_start, progress_percent_end, reports_count = 0.0, 0.01, 5
+        progress = Progress(self, progress_percent_start, progress_percent_end, reports_count)
+        input_workspace = self.getProperty('InputWorkspace').value
+        adjustment_diagnostics = list()  # list workspace names that analyze the orientation of the banks
+
+        # Create a grouping workspace whereby we group detectors by banks
+        grouping_workspace = self.temp_ws()  # a temporary name
```

Comment:
Just a comment here, `self.temp_ws` is not a workspace, but a workspace generator.  Future developers might have to spend a little bit extra time to figuring this part out.